File: hackerrank/30-days-of-code/day6-lets-review.js
function processData(input) {
    input = input.split('\n');

    input[0] = Number(input[0]);
    while (input[0] > 0) {
        const even = [], odd = [];
        const tmp = input.length - input[0];

        for (let i = 0; i < input[tmp].length; i++) {
            if (i % 2 === 0) {
                even.push(input[tmp].charAt(i));
            } else {
                odd.push(input[tmp].charAt(i));
            }
        }
        console.log(even.join(''), odd.join(''));
        input[0]--;
    }
}

File: hackerrank/30-days-of-code/day6-lets-review.java
import java.io.*;
import java.util.*;

public class Solution {

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int T = in.nextInt();
        String s;
        in.nextLine();

        ArrayList<Character> odd = new ArrayList<Character>();
        ArrayList<Character> even = new ArrayList<Character>();

        while (T > 0) {
            s = in.nextLine();

            for (int i = 0; i < s.length(); i++) {
                if (i % 2 == 0) {
                    even.add(s.charAt(i));
                } else {
                    odd.add(s.charAt(i));
                }
            }

            for (int i = 0; i < even.size(); i++) {
                System.out.print(even.get(i));
            }
            System.out.print(" ");
            for (int i = 0; i < odd.size(); i++) {
                System.out.print(odd.get(i));
            }
            System.out.println();

            odd.clear();
            even.clear();
            T--;
        }
    }
}
